### PR TITLE
Vacuum sqlite database after removing all data

### PR DESCRIPTION
**Pull request by mikakruschel**
Bodega now calls `vacuum` on the SQLite database after removing all data. While vacuuming after deleting all data is useful, I don't think it's absolutely necessary. That's why I decided not to throw any errors from vacuuming. I also added a test to ensure this actually works.

### CODE REVIEW

File: Sources/Bodega/SQLiteStorageEngine.swift [hunk 1]
```diff
@@ -211,6 +211,7 @@ public actor SQLiteStorageEngine: StorageEngine {
     /// Removes all the `Data` items from the database.
     public func removeAllData() throws {
         try self.connection.run(Self.storageTable.delete())
+        let _ = try? self.connection.vacuum()
```
Comment (mergesort) [inline Sources/Bodega/SQLiteStorageEngine.swift:214]: @mikakruschel It appears that `vacuum()` is annotated as `@ discardableResult`, which means we can safely shorten this to `try? self.connection.vacuum()`. Would you be able to make that small change?

File: Tests/BodegaTests/SQLiteStorageEngineTests.swift [hunk 2]
```diff
@@ -195,6 +195,25 @@ final class SQLiteStorageEngineTests: XCTestCase {
         XCTAssertEqual(updatedKeyCount, 0)
     }
 
+    func testVacuumAfterRemoveAllData() async throws {
```
Comment (mergesort) [inline Tests/BodegaTests/SQLiteStorageEngineTests.swift:198]: This test is correct, but would you mind making a few changes to match the other tests?

1. I would prefer to use the `storage` property defined at the top of the file, rather than creating `otherStorage` below. 
2. I would also prefer to use the `writeItemsToDatabase` function rather than creating a for loop in line to write items to the database. You're also free to update the `writeItemsToDatabase` method to take in an SQLiteStorageEngine if that would help simplify the code.
3. I didn't test this and it's more of a nit but I think you can skip the explicit `: Int` type annotations to match the style elsewhere.

**Review by mergesort — CHANGES_REQUESTED** [review]
@mikakruschel This looks great, there are just a couple of small changes I'd love to see before I merging in the pull request. Thank you so much!

File: Tests/BodegaTests/SQLiteStorageEngineTests.swift (hunk 2 above)
Comment (mikakruschel) [reply Tests/BodegaTests/SQLiteStorageEngineTests.swift:198]: The main problem with this test, as I wrote it, compared to the other tests, is that it requires the location of the sqlite3 file. However, since this location is private in `SQLiteStorageEngine`, I cannot access it in the test. Therefore, I have to create a new storage and store the location. Another possible solution, which may be better, is to store the temporary path of `storage` in the `SQLiteStorageEngineTests` class. Will change that.

Comment (mikakruschel) [reply Tests/BodegaTests/SQLiteStorageEngineTests.swift:202]: I've updated the test, and I think it looks much better now.

**Comment by mergesort** [comment]
Hey @mikakruschel, sorry it took me a bit to look at this, for some reason Github had marked my inbox as empty so I forgot about this.

I downloaded your patch and found that the tests were broken with some duplicate properties and a missing `}` to close out the file. I fixed the errors and cleaned up the code a little bit to my personal preferences, and now all of the tests pass. If this looks sufficient for you I'd be happy to merge in the changes, want to let me know what you think?

Thanks again for the idea and the work!
